identity-iota/src/credential/errors.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::fmt::Display;
+
+use identity_core::common::OneOrMany;
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+/// An error associated with validating credentials and presentations.
+pub enum ValidationError {
+  /// Indicates that the expiration date of the credential is not considered valid.
+  #[error("the expiration date is in the past or earlier than required.")]
+  ExpirationDate,
+  /// Indicates that the issuance date of the credential is not considered valid.
+  #[error("the credential is yet to be active or has not been active for the required period.")]
+  IssuanceDate,
```

Comment:
Nit: error message should mention issuance date.

```suggestion
  #[error("issuance date is in the future or later than required")]
  IssuanceDate,
```